packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -59,19 +63,32 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 					? getBlockType( firstBlockName )?.icon
 					: stack;
 			}
+
+			const reusableTitle =
+				isReusableBlock( blocks[ 0 ] ) &&
```

Comment:
We have an `_isSingleBlockSelected` check above we could move the code to depending on whether we end up with a general function for getting the reusable block name